packages/core/src/Table/TableRow.js

```diff
@@ -15,13 +17,29 @@ const tableRowStyles = css`
  * @example import { TableRow } from '@dhis2/ui-core'
  * @see Live demo: {@link /demo/?path=/story/table--static-layout|Storybook}
  */
-export const TableRow = ({ children, className, dataTest }) => (
-    <tr className={className} data-test={dataTest}>
-        {children}
+export const TableRow = ({
+    children,
+    className,
+    dataTest,
+    suppressZebraStriping,
+}) => {
+    const {
+        suppressZebraStriping: suppressZebraStripingFromContext,
+    } = useContext(TableContext)
 
-        <style jsx>{tableRowStyles}</style>
-    </tr>
-)
+    const zebraStriping =
+        typeof suppressZebraStriping !== 'undefined'
+            ? !suppressZebraStriping
+            : !suppressZebraStripingFromContext
```

Comment:
I would expect a table to have (or not have) zebra striping in its entirety. Is there a reason that the setting can be overridden with a prop on an individual TableRow as well? I would expect users to always use the TableRow with our Table.

Reply:
It's because we've defined that we won't use context unless it's overwritable by props and that props always have priority over context. @varl didn't want to have context in the core library at first until we agreed on the compromise to allow context as long as it can be overwritten with props.

Reply:
Ah, I didn't know that. Ok, I understand the intention. It keeps our components under user control at the component-level if they want that.

I do think that it makes for a bit of a confusing api though. It feels like we're using context to allow our components to communicate, so that we can have a simple top-level api on the Table component. Feels like that design intention runs kind of contrary to having components that are completely isolated. A bit like we're designing two different types of APIs.

I think we have that issue in some of our other components as well. Maybe we should discuss that on a more global level. Maybe you could say that one is the expected usage, and the other an escape hatch for more experienced devs? Though I wonder in what situation a dev would want to override the context and set the prop themselves.